client/my-sites/domains/domain-management/controller.jsx

```diff
@@ -72,6 +72,17 @@ export default {
 	},
 
 	domainManagementEdit( pageContext, next ) {
+		const selectedDomainName = decodeURIComponentIfValid( pageContext.params.domain );
+		if (
+			selectedDomainName &&
+			( selectedDomainName.endsWith( '.wordpress.com' ) ||
+				selectedDomainName.endsWith( '.wpcomstaging.com' ) )
```

Comment:
It works for both simple and atomic sites, but it doesn't work for free `.blog` subdomains (e.g. `rafagalani.home.blog`).

Maybe moving the logic and adding a `useEffect` function call on the inner `Settings` component could do it:

```javascript
useEffect( () => {
	if ( domain && isFreeUrlDomain( domain ) ) {
		page.redirect( domainManagementList( selectedSite.slug ) );
	}
}, [ domain, selectedSite ] );
```

Some more validations might be necessary, but something like it 😄

Reply:
I'd rather not load the component and the make the endpoint calls necessary to fetch this information just to do a simple redirect when.  Better to just use a utility function that looks for these well-known strings.